### ISSUE TITLE
Invalid no-js-extension

### ISSUE THREAD

**Issue by obrejla** [cvisco/eslint-plugin-requirejs#62]
This rule should not be applied for paths which have e.g. text! or template! prefix. It is intended to use it that was.

This should not produce error:

Text loading:

``` js
require(['text!/aaa/bbb/whatever.js'], function(template) {
    /* ... */
});
```

Template loading:

``` js
require(['template!/aaa/bbb/whatever.js'], function(template) {
    /* ... */
});
```

The solution could be to add an option for that rule with a list of "prefixes".

E.g.:
[2, ["text", "template"]]

**Comment by obrejla** [comment]
I can also try to fix it...

**Comment by platinumazure** [comment]
:+1: from me, I would say we should allow extensions for all plugins by default (only forbidding the ordinary module case) and allowing an option for what plugins the rule could be enforced on.

**Comment by obrejla** [comment]
Yep, it makes sense. So by default, everything prepended with something what matches this: `/^\w+\!/i` will be skipped, but when something is set in a options array and it will match, then it will not be skipped and will be checked.

**Comment by obrejla** [comment]
I started.
